test/functional/zapwallettxes.py

```diff
@@ -59,6 +59,7 @@ def run_test(self):
         self.start_node(0, ["-persistmempool=1", "-zapwallettxes=2"])
 
         wait_until(lambda: self.nodes[0].getmempoolinfo()['size'] == 1, timeout=3)
+        self.nodes[0].syncwithvalidationinterfacequeue()  # Flush mempool to wallet
 
         assert_equal(self.nodes[0].gettransaction(txid1)['txid'], txid1)
         assert_equal(self.nodes[0].gettransaction(txid2)['txid'], txid2)
```

Comment:
For the reviewer, the issue that it fixes here is: `JSONRPCException: Invalid or non-wallet transaction id (-5)`